### PR TITLE
AutoStyle integration into ok-client

### PR DESCRIPTION
**Pull request by ACscooter**
hi friends

### CODE REVIEW

File: client/protocols/autostyle.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import webbrowser
+
+log = logging.getLogger(__name__)
+
+
+class AutoStyleProtocol(models.Protocol):
+
+    # Timeouts are specified in seconds.
+    SHORT_TIMEOUT = 10
+    RETRY_LIMIT = 5
```
Comment (Sumukh) [inline client/protocols/autostyle.py]: Is this variable used?

File: client/protocols/autostyle.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
+import pdb
```
Comment (Sumukh) [inline client/protocols/autostyle.py]: This should be removed.

File: client/cli/ok.py [hunk 3]
```diff
@@ -65,8 +65,12 @@ def parse_input(command_input=None):
 
     # Autostyle
     parser.add_argument('--style', action='store_true',
+<<<<<<< HEAD
+                        help="run autostyle procedure")
+=======
                         help="Run AutoStyle Feedback System")
 
+>>>>>>> d37fce33ba5bbd105b0706d8105c4d0e132b04d5
```
Comment (knrafto) [inline client/cli/ok.py]: Merge conflict???

File: client/protocols/autostyle.py [hunk 4]
```diff
@@ -1,11 +1,23 @@
...
 
 log = logging.getLogger(__name__)
 
+
 class AutoStyleProtocol(models.Protocol):
+
+    # Timeouts are specified in seconds.
+    SHORT_TIMEOUT = 10
+    API_ENDPOINT = '{prefix}://{server}'
+    ALLOW_QUESTIONS = ['flatten', 'add_up', 'permutations']
```
Comment (Sumukh) [inline client/protocols/autostyle.py]: Q: Are these the only questions for the foreseeable future?

Comment (Sumukh) [inline client/protocols/autostyle.py]: Answer from @ACscooter: Yes - for now.

File: client/protocols/autostyle.py [hunk 5]
```diff
@@ -17,9 +29,79 @@ def run(self, messages):
...
+        data = {
+            'assignment': self.assignment.endpoint,
+            'messages': messages,
+            'submit': self.args.submit
+        }
+        serialized_data = json.dumps(data).encode(encoding='utf-8')
+        server = 'codestyle.herokuapp.com/ok_launch/'
 
-        messages['analytics']['autostyle'] = 'not ready'
+        address = self.API_ENDPOINT.format(server=server, prefix='http' if self.args.insecure else 'https')
```
Comment (Sumukh) [inline client/protocols/autostyle.py]: nit: This is probably better off as it's own variable (108 characters is a bit long)

**Comment by Sumukh** [comment]
LGTM. 

A few lines are over 80 characters but that seems ok for now.

**Pull request merged**